src/algorithms/calorimetry/HEXPLIT.cc

```diff
@@ -0,0 +1,153 @@
+// Copyright (C) 2023 Sebouh Paul
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+// References:
+//   https://arxiv.org/abs/2308.06939
+
+#include <DD4hep/Alignments.h>
+#include <DD4hep/DetElement.h>
+#include <DD4hep/Objects.h>
+#include <DD4hep/VolumeManager.h>
+#include <Evaluator/DD4hepUnits.h>
+#include <Math/GenVector/Cartesian3D.h>
+#include <Math/GenVector/DisplacementVector3D.h>
+#include <edm4hep/Vector3f.h>
+#include <stdlib.h>
+#include <algorithm>
+#include <cmath>
+#include <gsl/pointers>                            // for not_null
+#include <vector>
+#include "HEXPLIT.h"
+#include "algorithms/calorimetry/HEXPLITConfig.h"
+
+namespace eicrecon {
+
+const double HEXPLIT::neighbor_offsets_x[NEIGHBORS]={1.5*cos(0), 1.5*cos(M_PI/3), 1.5*cos(2*M_PI/3),1.5*cos(3*M_PI/3), 1.5*cos(4*M_PI/3), 1.5*cos(5*M_PI/3),
+    -sqrt(3)/2.*sin(0),-sqrt(3)/2.*sin(M_PI/3),-sqrt(3)/2.*sin(2*M_PI/3),-sqrt(3)/2.*sin(3*M_PI/3),-sqrt(3)/2.*sin(4*M_PI/3),-sqrt(3)/2.*sin(5*M_PI/3)};
+const double HEXPLIT::neighbor_offsets_y[NEIGHBORS]={1.5*sin(0), 1.5*sin(M_PI/3), 1.5*sin(2*M_PI/3),1.5*sin(3*M_PI/3), 1.5*sin(4*M_PI/3), 1.5*sin(5*M_PI/3),
+                          sqrt(3)/2.*cos(0), sqrt(3)/2.*cos(M_PI/3), sqrt(3)/2.*cos(2*M_PI/3), sqrt(3)/2.*cos(3*M_PI/3), sqrt(3)/2.*cos(4*M_PI/3), sqrt(3)/2.*cos(5*M_PI/3)};
```

Comment:
You could fill these programmatically with a lambda. Something like:
```suggestion
const double HEXPLIT::neighbor_offsets_x[NEIGHBORS] =[](){
  std::vector<double> x;
  std::vector<double> rs{1.5, -sqrt(3)/2.};
  for (auto& r : rs)
    for (theta = 0; theta < 2 * M_PI; theta += M_PI/3)
      x.push_back(r*cos(theta));
  return x;
}();
```